astroplan/plots/plots.py

```diff
@@ -0,0 +1,294 @@
+# (jazmin.berlanga@gmail.com), with input from
+# Adrian Price-Whelan (email), Erik Tollerud (email),
+# Christoph Deil (email), Eric Jeschke (email) and Brett Morris (email).
+
+
+# Replace function call later with below.
+def plot_airmass(target, observer, time, ax=None, style_kwargs=None, dark_plot=False):
+    """
+    TODO:
+        1) Timezones?
+        2) Limit airmass <= 3.
+        3) Plot lines instead of points.
+        4) Legend should not be transparent.
+        5) Airmass equation? Should this be moved to a separate function?
+        6) dark_plot option
+
+    Returns an ax object with an airmass vs. time plot.
+
+    If an ax object is passed in, plots an additional airmass plot on top.
+    Otherwise, creates a new ax object with an airmass plot.
+
+    Parameters
+    ----------
+    target: `~astroplan.FixedTarget`
+        The celestial body of interest.
+
+    observer: `~astroplan`
+        The person, telescope, observatory, etc. doing the observing.
+
+    time : `~astropy.time.Time`
+        An collection of times to use in generating data to plot.
+        Can be scalar (len(time)=1) or contain multiple times.
+
+    ax : `~matplotlib.axes.Axes` or None
+        Optional.
+
+    style_kwargs : WHAT TYPE IS DICTIONARY? or None
+        Optional.
+        A dictionary with plotting style options.
+
+    dark_plot: Boolean.
+        Optional. False (default) or True.
+    """
+
+    # Set up plot axes and style if needed.
+    if ax is None:
+        ax = plt.gca()
+    if style_kwargs is None:
+        # Need to add more style options for ax's with existing plots?
+        style_kwargs = {'linestyle': '-', 'color': 'r'}
+    #if dark_plot is True:
+        # Set dark plot style stuff here.
+
+    # Populate time window if needed.
+    if time.isscalar:
+        time = time + np.linspace(-12, 12, 100)*u.hour
+
+    # Calculate airmass
+    altitude = observer.altaz(time, target).alt
+    airmass = (1.0/np.cos(90*u.deg - altitude))
+
+    # Some checks & info for labels.
+    assert len(time) == len(airmass)
+
+    if hasattr(observer, 'name') is False:
+        observer_name = 'No Observer/Location Name'
+    else:
+        observer_name = observer.name
+
+    if hasattr(target, 'name') is False:
+        target_name = str('%.3f' % (random.random()))
+    else:
+        target_name = target.name
+
+    observe_date = time[0].datetime.strftime('%Y-%m-%d')
+    observe_timezone = 'UTC'
+
+    # Plot data.
+    ax.plot_date(
+        time.plot_date,
+        airmass,
+        linestyle=style_kwargs['linestyle'],
+        color=style_kwargs['color'],
+        label=target_name,
+        )
+    ax.figure.autofmt_xdate()
+    ax.invert_yaxis()
+
+    # Set labels, title, legend, etc.
+    ax.set_ylabel("Airmass")
+    ax.set_xlabel("Hour - "+observe_timezone)
+    ax.set_title("Airmass vs Time | "+observer_name+" | "+observe_date+" "+observe_timezone)
+    ax.legend(shadow=True)
```

Comment:
I don't think automatically adding the legend is a good idea: this plot only creates a single plot, so the legend is a bit superfluous.  The user can always call ``ax.legend(...)`` themselves later to add a legend if they want it.

(Same below).